Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -428,7 +419,15 @@ class HTTP1ConnectionProvider {
                 }
                 return self.state.offer(connection: connection)
             }
-            waiter.promise.succeed(connection)
+
+            switch action {
+            case .closeAnd:
+                // This happens when client was shut down during connect
```

Comment:
I think we should log in the failure case, no?

Reply:
good idea, done!